validphys2/src/validphys/plotutils.py

```diff
@@ -25,13 +26,59 @@
 
 log = logging.getLogger(__name__)
 
+def subplots(figsize=None,nrows=1, ncols =1,sharex=False, sharey=False):
+    """
+    Use matplotlib.figure.Figure() objects and not import pyplot anywhere. 
+    The reason is that pyplot maintains a global state that makes it misbehave 
+    in multithreaded applications such when executed under dask parallel mode.
+
+    Parameters
+    ----------
+    figsize : 2-tuple of floats
+            defaults is None
+    
+    nrows, ncols : int, default 1
+
+    sharex, sharey : bool, default False
+
+    Returns
+    -------
+    tuple
+        fig, ax = (matplotlib.figure.Figure, fig.subplots)
+    """
+    fig = Figure(figsize=figsize)
+    ax = fig.subplots(nrows=nrows, ncols = ncols, sharex=sharex, sharey=sharey)
+    return fig, ax
+
+def add_subplot(figsize=None,projection=None):
```

Comment:
I find the name of this function confusing because it not only adds a subplot but also creates a figure.
I naively expected it (when I saw it called before) to take a `Fig` as an input and add a subplot to that input figure.